bpfassets/perf_event/perf_event.c

```diff
@@ -206,30 +207,26 @@ int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
 
     // store process metrics
     struct process_metrics_t *process_metrics;
+    process_metrics = processes.lookup(&prev_pid);
+    if (process_metrics != 0)
+    {
+        // update process time
+        process_metrics->process_run_time += on_cpu_time_delta;
+
+        process_metrics->cpu_cycles += on_cpu_cycles_delta;
+        process_metrics->cpu_instr += on_cpu_instr_delta;
+        process_metrics->cache_miss += on_cpu_cache_miss_delta;
+    }
+
     process_metrics = processes.lookup(&cur_pid);
```

Comment:
I believe there is no need to update this. The only necessary update was to change the process ID to use the previous task. It's important to note that we should only update the metrics for the context of the previous task, as we were only collecting metrics for that task. There will be no data from the current task, so there is no need to add it to the hash at this time. Note that adding it to the hash would only cause the metrics to accumulate for the next context switch when it becomes the previous task.

Given this information, it is worth noting that we may encounter a loss of data with the new logic as it does not incorporate the current data of the previous task into the hash. As we have done previously, a new instance should be created within the hash.

Reply:
The new logic doesn't change the way `prev_pid` gets updated: as long as the `prev_pid` is there, its HW counters and cpu time are updated. 

The change only applies for `new_pid` (i.e. not previously seen before). In this case, since the new_pid has not started, only the pid and cgroup are recorded.